### DISCUSSION TITLE
Cannot start MyTourbook since I updated to 23.10.0!

### DISCUSSION THREAD

**Discussion by Wuffimaus** [mytourbook/mytourbook#1228] [General]
Hi,
I used old version of 18.xx.x or 19.xx.x - I can't remember. Now I have updated to 23.10.0, now I can't start MyTourBook anymore :-(
Here is the log file:

_> !SESSION Fri Nov 17 17:15:15 CET 2023 ------------------------------------------
!ENTRY org.eclipse.equinox.launcher 4 0 2023-11-17 17:15:15.932
!MESSAGE Could not find extension: reference:file:org.eclipse.osgi.compatibility.state_1.1.300.v20181116-1550.jar
!ENTRY org.eclipse.equinox.launcher 4 0 2023-11-17 17:15:15.954
!MESSAGE Exception launching the Eclipse Platform:
!STACK
java.lang.ClassNotFoundException: org.eclipse.core.runtime.adaptor.EclipseStarter
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)_

What did I wrong? I use Java 8.391 (Build 1.8.0_391-b13)!

Thanks and greetings from Germany!

Wuffi

**Comment by FJBDev** [comment]
@Wuffimaus 
MT requires Java 17 since v23.8 but you are running Java 8 🦕 
If you don't want to have to install Java separately, you can download 23.10 with Java embedded in the installer

You will see [here](https://sourceforge.net/projects/mytourbook/files/MyTourbook/23.10.0/) a file called `mytourbook-23.10.0-win-setup-with-java-runtime.exe`

I hope this helps!

**Comment by Wuffimaus** [comment]
I see, thanks! Maybe it's better do deinstall Java8 before? I'm confuse now, because 8 it's still newest, release last month at October:
https://www.java.com/de/download/

**Reply by FJBDev** [reply]
> I'm confuse now, because 8 it's still newest

Java 8 is still maintained but MyTourbook requires Java 17.

**Comment by wolfgang-ch** [comment]
I like these 🦖 🦕

**Comment by Wuffimaus** [comment]
Hi, I'm back, now at last I have finally time to continue here. 
I have installed now the newest MyTourbook from 2023-10-22 with Java! Now it's works 😊, but it's empty. ☹️
How can I restore or access my old database? Somewhere I could save a path, but I have forgotten how to do it again.
Here is my **old settings**:

_#This configuration file was written by: org.eclipse.equinox.internal.frameworkadmin.equinox.EquinoxFwConfigFileParser
#Tue Jun 04 16:38:41 CEST 2019
org.eclipse.update.reconcile=false
eclipse.p2.profile=DefaultProfile
osgi.instance.area.default=**C:\\Users\\Wuffi\\OneDrive\\Dokumente\\mytourbook**
osgi.framework=file\:plugins/org.eclipse.osgi_3.13.200.v20181130-2106.jar
equinox.use.ds=true
ds.delayed.keepInstances=true
osgi.bundles=reference\:file\:org.eclipse.equinox.simpleconfigurator_1.3.200.v20181101-1115.jar@1\:start
org.eclipse.equinox.simpleconfigurator.configUrl=file\:org.eclipse.equinox.simpleconfigurator/bundles.info
eclipse.product=net.tourbook.product
osgi.splashPath=platform\:/base/plugins/net.tourbook
osgi.framework.extensions=reference\:file\:org.eclipse.osgi.compatibility.state_1.1.300.v20181116-1550.jar
osgi.bundles.defaultStartLevel=4
eclipse.p2.data.area=@config.dir/../p2
eclipse.application=net.tourbook.application_

Thanks for your help!

**Reply by FJBDev** [reply]
>How can I restore or access my old database? Somewhere I could save a path, but I have forgotten how to do it again.

Did you restore your [previous backup](https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/backup) ?

**Comment by Wuffimaus** [comment]
As mentioned, the database is on Cloud :-).

![MyTourBook_WB_03](https://github.com/mytourbook/mytourbook/assets/151201330/22ddcb10-fe1a-4509-9d6d-071749ad0eec)

**Reply by FJBDev** [reply]
Did you try to specify the cloud path in the command line as below ?

 _"C:\Program Files\MyTourbook\mytourbook.exe" -data c:\mydata_

**Reply by wolfgang-ch** [reply]
The data directory is documented here https://mytourbook.sourceforge.io/mytourbook/index.php/download-install/data-directory

**Comment by Wuffimaus** [comment]
Vielen Dank Wolfgang, es hat geklappt 😊, Tourendatenbank von v38 auf v51 hat auch problemlos funktioniert! 
Allerdings muss ich immer über cmd wie folgt aufrufen:
mytourbook -data C:\Users\Wuffi\OneDrive\Dokumente\mytourbook

Das Verzeichnis würde ich gerne im config.ini hinterlegen, sieht in der neue Version V51 ganz anders aus als vorher:

_#This configuration file was written by: org.eclipse.equinox.internal.frameworkadmin.equinox.EquinoxFwConfigFileParser
#Sun Oct 22 07:01:52 CEST 2023
eclipse.product=net.tourbook.productWithJRE
osgi.splashPath=platform\:/base/plugins/net.tourbook
org.eclipse.equinox.simpleconfigurator.configUrl=file\:org.eclipse.equinox.simpleconfigurator/bundles.info
osgi.bundles.defaultStartLevel=4
eclipse.application=net.tourbook.appWithJRE
osgi.instance.area.default=@user.home/mytourbook
osgi.bundles=reference\:file\:org.eclipse.equinox.simpleconfigurator_1.4.200.v20221111-1340.jar@1\:start
eclipse.p2.data.area=@config.dir/../p2
eclipse.p2.profile=DefaultProfile
osgi.framework=file\:plugins/org.eclipse.osgi_3.18.400.v20230509-2241.jar
osgi.framework.extensions=reference\:file\:org.eclipse.osgi.compatibility.state_1.2.800.v20221116-1440.jar_

Wo genau kann ich hier ersetzen?

**Reply by wolfgang-ch** [reply]
Das Datenverzeichnis ist in ` osgi.instance.area.default=@user.home/mytourbook`

Die Datei `config.ini` wird aber bei jeder neuen Version von MT **überschrieben**, ist auch [hier](https://mytourbook.sourceforge.io/mytourbook/index.php/download-install/data-directory) dokumentiert

**Comment by Wuffimaus** [comment]
Das ist kein Problem, nach jedem Update kann ich ja Config.ini neu überschreiben 😊, oder läuft das Update jetzt automatisch?

**osgi.instance.area.default=@user.home/mytourbook**
durch 
**osgi.instance.area.default=C:\Users\Wuffi\OneDrive\Dokumente\mytourbook** ersetzt! 
Hat nicht geklappt, wie muss da genau da stehen?

**Reply by wolfgang-ch** [reply]
probier es mal mit `osgi.instance.area.default=file:/C:/Users/Wuffi/OneDrive/Dokumente/mytourbook`

**Comment by wolfgang-ch** [comment]
>Das ist kein Problem, nach jedem Update kann ich ja Config.ini neu überschreiben 😊, 

Du solltest nur die Zeile `osgi.instance.area.default=...` überschreiben sonst könnte es sein, dass die anderen Parameter in der Datei nicht mehr korrekt sind.


>oder läuft das Update jetzt automatisch?

Neue Haupt-Versionen wird es nicht als Update geben, nur kleinere.

**Comment by Wuffimaus** [comment]
Danke, es hat geklappt 😊, also file:/ muss noch davor 🙂

Was anderes, was bedeutet die Schattierung bei der Höhe? Die bekomme ich auch nicht weg!

![MyTourBook_WB_05](https://github.com/mytourbook/mytourbook/assets/151201330/792b306c-1cc5-44c4-99f8-d6d4a1083410)

**Comment by wolfgang-ch** [comment]
> Was anderes, was bedeutet die Schattierung bei der Höhe? Die bekomme ich auch nicht weg!

You should ask questions in english that all user can easily read it.

The dark area is propably the night section

**Comment by Wuffimaus** [comment]
Sorry, of course in english.
It's not a bug, it's feature 😊, good idea! Thanks and have a nice weekend!
